docs/types.rst

```diff
@@ -798,40 +798,42 @@ if the local variable itself is just a reference.
 Arrays
 ------
 
-Arrays can have a compile-time fixed size or they can be dynamic.
-The are few restrictions for the element, it can also be
-another array, a mapping or a struct. The general restrictions for
-types apply, though, in that mappings can only be used in storage
-and publicly-visible functions need parameters that are :ref:`ABI types <ABI>`.
+Arrays can have a compile-time fixed size, or they can have a dynamic size.
 
 An array of fixed size ``k`` and element type ``T`` is written as ``T[k]``,
-an array of dynamic size as ``T[]``. As an example, an array of 5 dynamic
-arrays of ``uint`` is ``uint[][5]`` (note that the notation is reversed when
-compared to some other languages). To access the second uint in the
-third dynamic array, you use ``x[2][1]`` (indices are zero-based and
-access works in the opposite way of the declaration, i.e. ``x[2]``
-shaves off one level in the type from the right).
+and an array of dynamic size as ``T[]``.
 
-Accessing an array past its end causes a revert. If you want to add
-new elements, you have to use ``.push()`` or increase the ``.length``
-member (see below).
+For example, an array of 5 dynamic arrays of ``uint`` as ``uint[][5]``,
+the notation is reversed compared to some other languages.
+
+Indices are zero-based, and access is in the opposite direction of the declaration.
+For example to access the second uint in the third dynamic array, use ``x[2][1]``,
+and to access the third dynamic array, use ``x[2]``
```

Comment:
Add: Again, if you have an array `T[5] a` for some type `T` that can also be an array, then `a[2]` will always have type `T`.

Reply:
`.` missing.